### PR TITLE
Feat/map controls

### PR DESCRIPTION
**Pull request by mariogiampieri**
## Description

In order to be able to test / interact with the map:
- Map modes are available via radio cards. `Erase` is currently disabled but I added it here so we can demonstrate where it will be.
- Active zone is accessed via color button instead of dropdown. This gets us closer to the current UI.
- Map styles are updated. This isn't as dynamic as I'd like it to be but I don't think it's as important as other tasks we need for the demonstration so I opened #23 . 
- add reset map functionality. we will need to square how this is synced with the backend but will address that in #22 . 

## Reviewers

- Primary: @raphaellaude 
- Secondary:

## Checklist

- [ ] Added/Updated related documentation (if applicable).
- [ ] Added/Updated related unit tests (if applicable).

## Screenshots (if applicable):
image

### CODE REVIEW

File: app/src/app/constants/layers.ts [hunk 1]
```diff
@@ -13,14 +12,59 @@ export const DEFAULT_PAINT_STYLE: ExpressionSpecification = [
...
 export const ZONE_ASSIGNMENT_STYLE: ExpressionSpecification = [
-  // based on zone feature state, set fill color
   "case",
-  ["==", ["feature-state", "zone"], 1],
-  "#0099cd",
-  ["==", ["feature-state", "zone"], 2],
-  "#ffca5d",
-  ["==", ["feature-state", "zone"], 3],
-  "#00cd99",
+  ["==", ["feature-state", "zone"], "#dd4425"],
```
Comment (raphaellaude) [inline app/src/app/constants/layers.ts:16]: O/question: curious if we can get the hex code from the radix color objects in order to directly tie the palettes from `app/src/app/constants/colors.ts` to this.

File: app/src/app/page.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 "use client";
 import { MapComponent } from "./components/Map";
-import SidebarComponent from "./components/Sidebar";
+import SidebarComponent from "./components/sidebar/Sidebar";
```
Comment (raphaellaude) [inline app/src/app/page.tsx:3]: good idea

File: app/src/app/components/sidebar/ResetMapButton.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  const handleClickResetMap = () => {
+    mapStore.setFreshMap(true);
+  };
+
+  return (
+    <button
+      onClick={handleClickResetMap}
+      className=" m-3 p-1.5 bg-white rounded-lg shadow-md"
+    >
```
Comment (raphaellaude) [inline app/src/app/components/sidebar/ResetMapButton.tsx:14]: I: Use radix themes [Button](https://www.radix-ui.com/themes/docs/components/button) for consistency + accessibility.

File: app/src/app/components/sidebar/MapModeSelector.jsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  return (
+    <Box>
+      <RadioCards.Root
+        defaultValue="default"
+        value={activeTool}
+        onValueChange={handleRadioChange}
+        columns={{ initial: "1", sm: "3" }}
+      >
+        {activeTools.map((tool) => (
+          <Flex key={`${tool.mode}-flex`} direction="column" width="100%">
```
Comment (raphaellaude) [inline app/src/app/components/sidebar/MapModeSelector.jsx:36]: I: by default `RadioCards` are flex column. Don't think you need this. See [example](https://www.radix-ui.com/themes/docs/components/radio-cards)

File: app/src/app/components/sidebar/MapModeSelector.jsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            <RadioCards.Item
+              value={tool.mode}
+              id={tool.mode}
+              disabled={tool.disabled}
+            >
+              {tool.label}
```
Comment (raphaellaude) [inline app/src/app/components/sidebar/MapModeSelector.jsx:42]: O: Low effort nice to have would be adding some [icons](https://www.radix-ui.com/icons)

File: app/src/app/constants/layers.ts (hunk 1 above)
Comment (raphaellaude) [inline app/src/app/constants/layers.ts:16]: H: Don't the zones need to be integers?

File: app/src/app/components/sidebar/ColorPicker.jsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  if (!colorArray) return null;
+  const handleRadioChange = (value) => {
+    console.log(
+      "setting accumulated geoids to old zone",
+      selectedZone,
+      "new zone is",
+      value
+    );
+    setZoneAssignments(selectedZone, accumulatedGeoids);
+    setSelectedZone(value);
```
Comment (raphaellaude) [inline app/src/app/components/sidebar/ColorPicker.jsx:33]: H: We're setting the selected zone to a hex color? Strongly thing we should set the zones to integers. The colors could get reshuffled and mess this up. Worst case if you're using integers the map will just look different. And the backend is expecting integer zones.

**Review by raphaellaude — CHANGES_REQUESTED** [review]
Only one blocking comment – we should go back to storing the zones as integers. UI is pretty–nice work!

File: app/src/app/constants/layers.ts (hunk 1 above)
Comment (mariogiampieri) [reply app/src/app/constants/layers.ts:16]: set dynamically now, much less brittle

Comment (mariogiampieri) [reply app/src/app/constants/layers.ts]: addressed per below

File: app/src/app/components/sidebar/ColorPicker.jsx (hunk 6 above)
Comment (raphaellaude) [reply app/src/app/components/sidebar/ColorPicker.jsx:33]: Any ints will work. Can do 1-indexed or 0-indexed

**Review by raphaellaude — APPROVED** [review]
Lgtm!

File: app/src/app/components/sidebar/MapModeSelector.jsx (hunk 4 above)
Comment (mariogiampieri) [reply app/src/app/components/sidebar/MapModeSelector.jsx:36]: good call, removed

File: app/src/app/components/sidebar/MapModeSelector.jsx (hunk 5 above)
Comment (mariogiampieri) [reply app/src/app/components/sidebar/MapModeSelector.jsx:44]: added!

**Comment by mariogiampieri** [comment]
addressed those non-breaking comments above and ready for another review- talk soon!

**Review by raphaellaude — APPROVED** [review]
Lgtm!

**Comment by mariogiampieri** [comment]
closes #18 and #23

**Pull request merged**